databroker/core.py

```diff
@@ -26,6 +26,69 @@
 from .intake_xarray_core.base import DataSourceMixin
 from .intake_xarray_core.xarray_container import RemoteXarray
 from collections import deque
+from typing import Iterator, Mapping, TypeVar
+
+K = TypeVar("K")
+V = TypeVar("V")
+
+
+class Frozen(Mapping[K, V]):
```

Comment:
If we are going to go down the typing route, we can pin this a bit more to `K` always being `str` and `V` might as well be `Any`.